lib/ignore/ignoreresult/ignoreresult.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2024 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Package result provides the result type for ignore matching. This is a
+// separate package in order to break import cycles.
+package ignoreresult
+
+import "github.com/syncthing/syncthing/lib/build"
+
+const (
+	NotMatched R = 0
+
+	// The bit flags are used by the ignore package to construct Results.
+	// Don't use them for comparison; use the convenience methods.
```

Comment:
Could we get setters or a constructor for the result to use in ignore to avoid this?

Reply:
We can!